File: calcTax.py
def calcTax(levels,salary):
    tax = 0.0
    left = salary
    limit = 0.0
    taxableIncome = 0
    i=0
    while (left > 0):
        level = levels[i]
        pecent = level[1]
        if (level[0] == None):
            tax += left * pecent
            return tax
        taxableIncome = min(level[0] - limit, left)
        tax += taxableIncome*pecent
        limit = level[0]
        left = left - taxableIncome
        i = i + 1
    return tax

levels = [[1000.0,0.1],[2000.0,0.2],[3000.0,0.3]]
calcTax(levels,5000)

File: Calc_Tax.java
// Time Complexity :O(n)
// Space Complexity :O(1)
// Did this code successfully run on Leetcode :No(Glassdoor question )
// Any problem you faced while coding this :
//https://www.glassdoor.co.in/Interview/Write-a-program-in-Python-to-calculate-tax-if-Salary-and-Tax-Brackets-are-given-as-list-in-the-form-10000-3-20000-QTN_2565898.htm

// Your code here along with comments explaining your approach
import java.io.*;

import java.util.*;

class GFG {

	public static void main (String[] args) {

		List<List<Double>> levels = new ArrayList<>();

        levels.add( Arrays.asList(10000.0, 0.1) );

        levels.add( Arrays.asList(20000.0, 0.2) );

        levels.add( Arrays.asList(30000.0, 0.3) );

        levels.add( Arrays.asList(null, 0.4) );

        double tax = calculateTax(levels,45000);

        System.out.println(tax);

	}

        public static double calculateTax(List<List<Double>> levels, double salary)
        {
            
            double salaryLeft = salary;

            double Totaltax = 0;

            double prevLimit = 0, percent = 0, currLimit = 0, currRange = 0, currSalary = 0;
            
            
            int i = 0;
            while(salaryLeft>0)
            {
                List<Double> level = levels.get(i);

                i++;

                percent = level.get(1);
                
                
                if(level.get(0) == null){

                    Totaltax += percent* salaryLeft;
                    return Totaltax;
                }
                currLimit = level.get(0);
                
                currRange = currLimit - prevLimit;

                currSalary = Math.min(salaryLeft, currRange);

                Totaltax += currSalary*percent;

                salaryLeft -= currSalary;

                prevLimit = currLimit;
            }

            return Totaltax;
        }

}